stix_shifter_modules/sentinelone/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,197 @@
+                else:
+                    cursor = None
+
+                while cursor is not None:
+                    response = self.api_client.get_search_results(
+                        search_id, min_range, max_range, nextcursor=cursor)
+                    response_code = response.code
+                    response_txt = response.read().decode('utf-8')
+                    response_dict = json.loads(response_txt)
+                    response_dict = self.check_empty_data(response_dict)
+
+                    for item in response_dict['data']:
+                        total_data.append(item)
+
+                    if response_dict.get("pagination"):
+                        pagination_dict = response_dict.get("pagination")
+                        if pagination_dict.get("nextCursor"):
+                            cursor = pagination_dict.get("nextCursor")
+                        else:
+                            cursor = None
+
+            if response_code == 200:
+                return_obj['success'] = True
+                return_obj['data'] = total_data[min_range:max_range] if total_data else []
+            elif response_code == 400:
+                return_obj['success'] = False
+                response_code = response_dict.get("errors")[0].get("code")
+                if response_code == 4000010:
+                    raise LimitOutOfRangeError
+            else:
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+
+        except ConnectionError:
+            response_dict['type'] = "ConnectionError"
+            response_dict['message'] = "Invalid Host"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        except LimitOutOfRangeError:
+            response_dict['type'] = "LimitOutOfRangeError"
+            response_dict['message'] = "Limit must be greater than or equals to 1 " \
+                                       "and less than equals to 1000"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        except QueryIdNotFoundError:
+            response_dict['type'] = "QueryIdNotFoundError"
+            response_dict['message'] = "Could not find query id: " + search_id
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        except Exception as ex:
+            self.logger.error('error in query result: %s', str(ex))
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        return return_obj
+
+    @staticmethod
+    def check_empty_data(res_dict):
+        """
+        Format the results in json format
+        :param item: list of dictionary items
+        :return: list
+        """
+        for item in res_dict['data']:
+            if item.get("netProtocolName") is None:
+                if item.get("srcPort") is not None or item.get("dstPort") is not None:
+                    item["netProtocolName"] = "null"
```

Comment:
this doesn't make sense. why are you setting protocol name to null? I understand that to create `network-traffic` object but there must be default protocol from the api to use here or some sort of keyword. setting `null` value seems really odd.

Reply:
stix validation error was getting when field  netProtocolName had null value. There is no default protocol from api, so based on eventtype assigning "tcp" or "http" in place of "null".